### ISSUE TITLE
Cannot configure codemirror client functions

### ISSUE THREAD

**Issue by webteckie** [keystonejs/keystone#3038]
### Expected behavior

Codemirror allows configuring client functions for certain behaviors. When a list model configures a codemirror field type and configures such behavior functions the functions should be available for execution in the client side. 

### Actual behavior

The functions are not delivered to the client side.  Actually, this is a more general keystone issue with packing list model functions for delivery to the client side.  @JedWatson is aware of it and thinking hard about a solution that addresses the total problem.

### Steps to reproduce the behavior

For example, codemirror can be configured to handle key presses that perform certain behavior:
```
    testKeys: {
        type: Types.Code,
        codemirror: {
            extraKeys: {
                "F11": function(cm) {
                    console.log("pressed F11");
                },
                "Esc": function(cm) {
                    console.log("pressed Esc");
                }
            }
        },
    },
```

**Comment by wmertens** [comment]
With the webpack branch, one can use `require.ensure` with the required mode, and thanks to  https://webpack.github.io/docs/context.html that will dynamically load only the desired mode. So something like `{type: Types.Code, mode: 'javascript'}` would run e.g. ```require.ensure(`codemirror/mode/${mode}/${mode}`, () => this.setState({depsLoaded: true}))``` in componentDidMount.

**Comment by webteckie** [comment]
@wmertens don't see how your previous comment is applicable to solve the issue.  Does the webpack build allow for defined custom functions to run in the client?

**Comment by wmertens** [comment]
The webpack build allows packaging anything for dynamic loading. With some small changes, it should be able to dynamically load anything that is available at build time. So that means the modes, and anything else that ships with codemirror.

To run custom functions, they would have to be provided as strings to the field configuration.

**Comment by webteckie** [comment]
So does the mode thingy above currently works as such?  Can it do the same for themes?  How about addons?  Can we add a new option to the Code Field that configures addons?  For example:
```
    content: {
        type: Types.Code,
        addons:  ["display/fullscreen", "lint/javascript-lint", etc.]  // but note that fullscreen has both a .css and a .js
        codemirror: {
            mode: {name: "javascript"},
            theme: "yeti dracula",
            ...
       }
```

If this is done, then there's no need to custom build codemirror!

**Comment by wmertens** [comment]
Yes, that's possible. We'll probably have to special-case the css so you don't have to mention it, but it's a matter of translating the options into an array for `require.ensure` and webpack will do the rest.
